src/vs/workbench/contrib/mergeEditor/browser/mergeEditorInput.ts

```diff
@@ -114,7 +120,7 @@ export class MergeEditorInput extends AbstractTextResourceEditorInput implements
 	}
 
 	override async save(group: number, options?: ITextFileSaveOptions | undefined): Promise<IUntypedEditorInput | undefined> {
-		await (await this.resolve()).save();
+		await this._inputModel?.save(options);
```

Comment:
No need to resolve.

Reply:
`save` is only ever called after resolve succeeded?

Reply:
Yeah in principle. This really follows the exact same model we have for text files, that is why I am not so worried of doing this. Text files will return early if no associated model exists:

https://github.com/microsoft/vscode/blob/6d7beb43a0a3de03fe2b1ee53899416fac36154c/src/vs/workbench/services/textfile/browser/textFileService.ts#L348-L351